File: src/main/java/com/DavidM1A2/afraidofthedark/common/spell/component/powerSource/base/AOTDSpellPowerSource.java
package com.DavidM1A2.afraidofthedark.common.spell.component.powerSource.base;

import com.DavidM1A2.afraidofthedark.common.constants.Constants;
import net.minecraft.util.ResourceLocation;

/**
 * Base class for all AOTD spell power sources
 */
public abstract class AOTDSpellPowerSource extends SpellPowerSource
{
    /**
     * Constructor sets the power source's name
     *
     * @param name The name of the power source
     */
    public AOTDSpellPowerSource(String name)
    {
        super();
        this.setRegistryName(new ResourceLocation(Constants.MOD_ID, name));
    }
}

File: src/main/kotlin/com/davidm1a2/afraidofthedark/common/spell/component/powerSource/base/AOTDSpellPowerSource.kt
package com.davidm1a2.afraidofthedark.common.spell.component.powerSource.base

import com.davidm1a2.afraidofthedark.common.constants.Constants
import com.davidm1a2.afraidofthedark.common.research.Research
import net.minecraft.resources.ResourceLocation

abstract class AOTDSpellPowerSource<T>(name: String, prerequisiteResearch: Research? = null) : SpellPowerSource<T>(ResourceLocation(Constants.MOD_ID, name), prerequisiteResearch)
